kafkahook/kafkahook.go

```diff
@@ -39,6 +40,22 @@ type Config struct {
 	Producer  sarama.AsyncProducer
 }
 
+func NewWithContext(ctx context.Context, conf Config) (hook *KafkaHook, err error) {
+	done := make(chan struct{})
+	go func() {
+		hook, err = New(conf)
+		close(done)
+	}()
+
+	select {
+	case <-ctx.Done():
+		return hook, fmt.Errorf("connect timeout while connecting to kafka peers %s",
```

Comment:
It took me a few moment to realize that hook is nil here. It would be better IMHO not to waste those moments at all that can be achieved by returning nil explicitly `return nil, fmt.Err.....`